types/3box/index.d.ts

```diff
@@ -0,0 +1,116 @@
+    noAutoSub?: boolean;
+}
+export interface BoxSpace {
+    public: BoxKeyValueStore;
+    private: BoxKeyValueStore;
+    joinThread: (name: string, opts?: BoxSpaceOpts_joinThread) => BoxThread;
+    subscribeThread: (name: string) => undefined;
+    unsubscribeThread: (name: string) => undefined;
+    subscribedThreads: () => string[];
+    BoxSpace: () => BoxSpace;
+}
+export interface BoxKeyValueStoreMetadata {
+    // missing definition
+    dummy: boolean;
+}
+
+export interface BoxKeyValueStore {
+    BoxKeyValueStore: () => BoxKeyValueStore;
+    log: any[];
+    get: (key: any) => any;
+    getMetadata: (key: any) => BoxKeyValueStoreMetadata | undefined;
+    set: (key: any, value: any) => boolean;
+    remove: (key: any) => boolean;
+}
+export interface BoxIsMuportDIDReturn {
+    // missing definition
+    dummy: boolean;
+}
+export class BoxStaticIdUtils {
+    verifyClaim: () => any;
+    isMuportDID: (address: any) => BoxIsMuportDIDReturn | boolean;
+    isClaim: (claim: any, opts: any) => Promise<boolean>;
+}
+export interface BoxObjectOpts_OpenSpace {
+    consentCallback?: () => any;
+    onSyncDone?: () => any;
+}
+export interface BoxObjectOpts_GetProfile {
+    addressServer?: string;
+    ipfs?: any; // ipfs object
+    useCacheService?: boolean;
+    profileServer?: string;
+}
+export interface BoxObjectOpts_GetProfiles {
+    profileServer?: string;
+}
+export interface BoxObjectOpts_GetSpace {
+    profileServer?: string;
+    metadata?: string;
+}
+export interface BoxObjectOpts_GetThread {
+    profileServer?: string;
+}
+export interface BoxObjectOpts_ListSpaces {
+    profileServer?: string;
+}
+export interface BoxObjectOpts_profileGraphQL {
+    graphqlServer?: string;
+}
+export interface BoxObjectOpts_openBox {
+    consentCallback?: () => any;
+    pinningNode?: string;
+    ipfs?: any; // ipfsobj
+    addressServer?: string;
+}
+
+export class BoxInstance {
+    public: BoxKeyValueStore;
+    private: BoxKeyValueStore;
+    verified: BoxVerified;
+    spaces: any;
+    openSpace: (name: string, opts?: BoxObjectOpts_OpenSpace) => any;
+    onSyncDone: (cb: () => undefined) => null;
+    logout: () => any;
+}
+
+export const idUtils: BoxStaticIdUtils;
+export function getProfile(address: any, opts?: BoxObjectOpts_GetProfile): Promise<any>;
+export function getProfiles(address: any[], opts?: BoxObjectOpts_GetProfiles): undefined;
+export function getSpace(address: any, name: any, opts?: BoxObjectOpts_GetSpace): undefined;
+export function getThread(space: any, name: any, opts?: BoxObjectOpts_GetThread): string[];
+export function listSpaces(address: any, opts?: BoxObjectOpts_ListSpaces): undefined;
+export function profileGraphQL(query: any, opts?: BoxObjectOpts_profileGraphQL): undefined;
+export function getVerifiedAccounts(profile: any): undefined;
+export function openBox(address: any, ethereumProvider: any, opts?: BoxObjectOpts_openBox): BoxInstance;
+export function isLoggedIn(address: any): undefined;
```

Comment:
You would want to write `export as namespace` declaration as well since as per https://github.com/3box/3box-js#usage it can be used in html code as well. 
Refer [this](https://github.com/microsoft/TypeScript-Handbook/blob/master/pages/declaration%20files/Library%20Structures.md#umd) for more details.